frontend/src/components/Permissions.tsx

```diff
@@ -1,274 +1,235 @@
+  );
 
-    handlePermissionsChange({
-      scopes: newScopes,
-      budgetRenewal,
-    });
-  };
+  const handleScopeChange = React.useCallback(
+    (scopes: Set<Scope>) => {
+      handlePermissionsChange({ scopes });
+    },
+    [handlePermissionsChange]
+  );
 
-  const handleMaxAmountChange = (amount: number) => {
-    handlePermissionsChange({ maxAmount: amount });
-  };
+  const handleBudgetMaxAmountChange = React.useCallback(
+    (amount: number) => {
+      handlePermissionsChange({ maxAmount: amount });
+    },
+    [handlePermissionsChange]
+  );
 
-  const handleBudgetRenewalChange = (value: string) => {
-    handlePermissionsChange({ budgetRenewal: value as BudgetRenewalType });
-  };
+  const handleBudgetRenewalChange = React.useCallback(
+    (value: string) => {
+      handlePermissionsChange({ budgetRenewal: value as BudgetRenewalType });
+    },
+    [handlePermissionsChange]
+  );
 
-  const handleDaysChange = (days: number) => {
-    setDays(days);
-    if (!days) {
-      handlePermissionsChange({ expiresAt: undefined });
-      return;
-    }
-    const currentDate = new Date();
-    const expiryDate = new Date(
-      Date.UTC(
-        currentDate.getUTCFullYear(),
-        currentDate.getUTCMonth(),
-        currentDate.getUTCDate() + days,
-        23,
-        59,
-        59,
-        0
-      )
-    );
-    handlePermissionsChange({ expiresAt: expiryDate });
-  };
+  const handleExpiryChange = React.useCallback(
+    (expiryDate?: Date) => {
+      handlePermissionsChange({ expiresAt: expiryDate });
+    },
+    [handlePermissionsChange]
+  );
 
   return (
-    <div>
-      <div className="mb-6">
-        <ul className="flex flex-col w-full">
-          {capabilities?.scopes.map((scope, index) => {
-            const ScopeIcon = iconMap[scope];
-            return (
-              <li
-                key={index}
+    <div className="max-w-lg">
+      {isScopesEditable ? (
+        <Scopes
+          capabilities={capabilities}
+          scopes={permissions.scopes}
+          onScopeChange={handleScopeChange}
+        />
+      ) : (
+        <>
+          <p className="text-sm font-medium mb-2">Scopes</p>
+          <div className="flex flex-col mb-2">
+            {[...permissions.scopes].map((rm) => {
+              const PermissionIcon = iconMap[rm];
+              return (
+                <div
+                  key={rm}
+                  className={cn(
+                    "flex items-center mb-2",
+                    rm == NIP_47_PAY_INVOICE_METHOD && "order-last"
+                  )}
+                >
+                  <PermissionIcon className="mr-2 w-4 h-4" />
+                  <p className="text-sm">{scopeDescriptions[rm]}</p>
+                </div>
+              );
+            })}
+          </div>
+        </>
+      )}
+      {capabilities.scopes.includes(NIP_47_PAY_INVOICE_METHOD) &&
+        permissions.scopes.has(NIP_47_PAY_INVOICE_METHOD) &&
+        (!isBudgetAmountEditable ? (
```

Comment:
If budget amount is set and budget renewal isn't (in the params) then it assumes renewal as "Monthly" and the user can't change it. Avoiding option to edit budget renewal in these cases as that adds unnecessary complexity.

Reply:
Yeah, I believe default should be never. Budgets from a deeplink-created app are different than the default user flow, and monthly in this case does not make sense